bash-preexec.sh

```diff
@@ -47,6 +47,12 @@ if [[ -z "${BASH_VERSINFO-}" ]] || (( BASH_VERSINFO[0] < 3 || (BASH_VERSINFO[0]
     return 1
 fi
 
+# We do not enable bash-preexc in non-interactive shells (except in tests).
+# `__bp_inside_test` is set in test/*.bats in bash-preexec repository.
+if [[ $- != *i* && ! ${__bp_inside_test-} ]]; then
```

Comment:
Do you think we could use one of the [Bats special variables](https://bats-core.readthedocs.io/en/stable/writing-tests.html#special-variables) instead (e.g. `$BATS_VERSION`)?

Reply:
That's a good point. Maybe use `$BATS_TEST_TAGS` and tag first-party tests with a `bash-preexec` tag? That would also allow others to opt-in.

It wouldn't be that different from what you're doing now, but it maybe a more standard (Bats-specific) approach.

Reply:
That seems useful. I'll later check it.